Crunch docker build fails

I am trying to build the docker image with the provided `Dockerfile` in this repo, but the build fails. 
Note: tried both `v0.19.0` and `v0.20.0` on both Linux and Mac. 

## System Info of the Linux:
```
PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

## Steps to reproduce: 

```
git clone http://github.com/turboflakes/crunch
cd crunch
docker build -t localhost/crunch .
```


## Error logs
<details>

```
135.5    Compiling crunch v0.19.0 (/app)
174.6 error[E0283]: type annotations needed
174.6     --> src/runtimes/polkadot.rs:1569:67
174.6      |
174.6 1569 |                     if claimable > config.pool_compound_threshold.into() {
174.6      |                                  -                                ^^^^
174.6      |                                  |
174.6      |                                  type must be known at this point
174.6      |
174.6      = note: multiple `impl`s satisfying `u128: PartialOrd<_>` found in the following crates: `core`, `deranged`:
174.6              - impl PartialOrd for u128;
174.6              - impl<MIN, MAX> PartialOrd<deranged::RangedU128<MIN, MAX>> for u128
174.6                where the constant `MIN` has type `u128`, the constant `MAX` has type `u128`;
174.6 help: try using a fully qualified path to specify the expected types
174.6      |
174.6 1569 |                     if claimable > <u64 as Into<T>>::into(config.pool_compound_threshold) {
174.6      |                                    +++++++++++++++++++++++                              ~
174.6 
174.7 error[E0283]: type annotations needed
174.7     --> src/runtimes/kusama.rs:1568:67
174.7      |
174.7 1568 |                     if claimable > config.pool_compound_threshold.into() {
174.7      |                                  -                                ^^^^
174.7      |                                  |
174.7      |                                  type must be known at this point
174.7      |
174.7      = note: multiple `impl`s satisfying `u128: PartialOrd<_>` found in the following crates: `core`, `deranged`:
174.7              - impl PartialOrd for u128;
174.7              - impl<MIN, MAX> PartialOrd<deranged::RangedU128<MIN, MAX>> for u128
174.7                where the constant `MIN` has type `u128`, the constant `MAX` has type `u128`;
174.7 help: try using a fully qualified path to specify the expected types
174.7      |
174.7 1568 |                     if claimable > <u64 as Into<T>>::into(config.pool_compound_threshold) {
174.7      |                                    +++++++++++++++++++++++                              ~
174.7 
174.8 error[E0283]: type annotations needed
174.8     --> src/runtimes/paseo.rs:1562:67
174.8      |
174.8 1562 |                     if claimable > config.pool_compound_threshold.into() {
174.8      |                                  -                                ^^^^
174.8      |                                  |
174.8      |                                  type must be known at this point
174.8      |
174.8      = note: multiple `impl`s satisfying `u128: PartialOrd<_>` found in the following crates: `core`, `deranged`:
174.8              - impl PartialOrd for u128;
174.8              - impl<MIN, MAX> PartialOrd<deranged::RangedU128<MIN, MAX>> for u128
174.8                where the constant `MIN` has type `u128`, the constant `MAX` has type `u128`;
174.8 help: try using a fully qualified path to specify the expected types
174.8      |
174.8 1562 |                     if claimable > <u64 as Into<T>>::into(config.pool_compound_threshold) {
174.8      |                                    +++++++++++++++++++++++                              ~
174.8 
175.0 error[E0283]: type annotations needed
175.0     --> src/runtimes/westend.rs:1556:67
175.0      |
175.0 1556 |                     if claimable > config.pool_compound_threshold.into() {
175.0      |                                  -                                ^^^^
175.0      |                                  |
175.0      |                                  type must be known at this point
175.0      |
175.0      = note: multiple `impl`s satisfying `u128: PartialOrd<_>` found in the following crates: `core`, `deranged`:
175.0              - impl PartialOrd for u128;
175.0              - impl<MIN, MAX> PartialOrd<deranged::RangedU128<MIN, MAX>> for u128
175.0                where the constant `MIN` has type `u128`, the constant `MAX` has type `u128`;
175.0 help: try using a fully qualified path to specify the expected types
175.0      |
175.0 1556 |                     if claimable > <u64 as Into<T>>::into(config.pool_compound_threshold) {
175.0      |                                    +++++++++++++++++++++++                              ~
175.0 
175.0 error[E0283]: type annotations needed
175.0     --> src/runtimes/kusama.rs:1496:67
175.0      |
175.0 1496 |                     if claimable > config.pool_compound_threshold.into() {
175.0      |                                  -                                ^^^^
175.0      |                                  |
175.0      |                                  type must be known at this point
175.0      |
175.0      = note: multiple `impl`s satisfying `u128: PartialOrd<_>` found in the following crates: `core`, `deranged`:
175.0              - impl PartialOrd for u128;
175.0              - impl<MIN, MAX> PartialOrd<deranged::RangedU128<MIN, MAX>> for u128
175.0                where the constant `MIN` has type `u128`, the constant `MAX` has type `u128`;
175.0 help: try using a fully qualified path to specify the expected types
175.0      |
175.0 1496 |                     if claimable > <u64 as Into<T>>::into(config.pool_compound_threshold) {
175.0      |                                    +++++++++++++++++++++++                              ~
175.0 
179.0 error[E0283]: type annotations needed
179.0     --> src/runtimes/paseo.rs:1490:67
179.0      |
179.0 1490 |                     if claimable > config.pool_compound_threshold.into() {
179.0      |                                  -                                ^^^^
179.0      |                                  |
179.0      |                                  type must be known at this point
179.0      |
179.0      = note: multiple `impl`s satisfying `u128: PartialOrd<_>` found in the following crates: `core`, `deranged`:
179.0              - impl PartialOrd for u128;
179.0              - impl<MIN, MAX> PartialOrd<deranged::RangedU128<MIN, MAX>> for u128
179.0                where the constant `MIN` has type `u128`, the constant `MAX` has type `u128`;
179.0 help: try using a fully qualified path to specify the expected types
179.0      |
179.0 1490 |                     if claimable > <u64 as Into<T>>::into(config.pool_compound_threshold) {
179.0      |                                    +++++++++++++++++++++++                              ~
179.0 
183.0 error[E0283]: type annotations needed
183.0     --> src/runtimes/polkadot.rs:1497:67
183.0      |
183.0 1497 |                     if claimable > config.pool_compound_threshold.into() {
183.0      |                                  -                                ^^^^
183.0      |                                  |
183.0      |                                  type must be known at this point
183.0      |
183.0      = note: multiple `impl`s satisfying `u128: PartialOrd<_>` found in the following crates: `core`, `deranged`:
183.0              - impl PartialOrd for u128;
183.0              - impl<MIN, MAX> PartialOrd<deranged::RangedU128<MIN, MAX>> for u128
183.0                where the constant `MIN` has type `u128`, the constant `MAX` has type `u128`;
183.0 help: try using a fully qualified path to specify the expected types
183.0      |
183.0 1497 |                     if claimable > <u64 as Into<T>>::into(config.pool_compound_threshold) {
183.0      |                                    +++++++++++++++++++++++                              ~
183.0 
187.3 error[E0283]: type annotations needed
187.3     --> src/runtimes/westend.rs:1484:67
187.3      |
187.3 1484 |                     if claimable > config.pool_compound_threshold.into() {
187.3      |                                  -                                ^^^^
187.3      |                                  |
187.3      |                                  type must be known at this point
187.3      |
187.3      = note: multiple `impl`s satisfying `u128: PartialOrd<_>` found in the following crates: `core`, `deranged`:
187.3              - impl PartialOrd for u128;
187.3              - impl<MIN, MAX> PartialOrd<deranged::RangedU128<MIN, MAX>> for u128
187.3                where the constant `MIN` has type `u128`, the constant `MAX` has type `u128`;
187.3 help: try using a fully qualified path to specify the expected types
187.3      |
187.3 1484 |                     if claimable > <u64 as Into<T>>::into(config.pool_compound_threshold) {
187.3      |                                    +++++++++++++++++++++++                              ~
187.3 
302.3 For more information about this error, try `rustc --explain E0283`.
302.6 error: could not compile `crunch` (bin "crunch") due to 8 previous errors
------
Dockerfile:13
--------------------
  11 |     COPY . /app
  12 |     WORKDIR /app
  13 | >>> RUN /root/.cargo/bin/cargo build --$PROFILE --package crunch
  14 |     
  15 |     # ===== SECOND STAGE ======
--------------------
ERROR: failed to solve: process "/bin/sh -c /root/.cargo/bin/cargo build --$PROFILE --package crunch" did not complete successfully: exit code: 101
```
<details>

--- a/Dockerfile
+++ b/Dockerfile
@@ -25,7 +25,7 @@ COPY --from=builder /app/target/$PROFILE/crunch /usr/local/bin
 RUN useradd -u 1000 -U -s /bin/sh crunch
 USER crunch
 
-ENV RUST_BACKTRACE 1
+ENV RUST_BACKTRACE=1
 ENV RUST_LOG="info"
 
 RUN /usr/local/bin/crunch --version

--- a/README.md
+++ b/README.md
@@ -517,10 +517,10 @@ The config and seed files can be mounted from the host. Any options supported by
 
 ```bash
 docker run --rm -it \
-  --volume=/etc/crunch/env:/.env:U \
-  --volume=/etc/crunch/private.seed:/.private.seed:U \
+  --volume=/etc/env:/.env:ro \
+  --volume=/etc/private.seed:/.private.seed:ro \
   localhost/crunch:latest \
-    rewards era
+    kusama rewards era
 ```
 
 ## Inspiration

--- a/src/crunch.rs
+++ b/src/crunch.rs
@@ -54,8 +54,10 @@ pub type ValidatorIndex = Option<usize>;
 pub type ValidatorAmount = u128;
 pub type NominatorsAmount = u128;
 
+#[allow(dead_code)]
 type Message = Vec<String>;
 
+#[allow(dead_code)]
 trait MessageTrait {
     fn log(&self);
     fn show_or_hide(&mut self, value: String, hidden: bool);
@@ -531,6 +533,7 @@ pub async fn try_fetch_stashes_from_remote_url(
 }
 
 #[derive(Deserialize, Clone, Debug)]
+#[allow(dead_code)]
 pub struct OnetData {
     pub address: String,
     pub grade: String,

--- a/src/report.rs
+++ b/src/report.rs
@@ -100,6 +100,7 @@ impl Validator {
 pub type Validators = Vec<Validator>;
 
 #[derive(Debug, Clone)]
+#[allow(dead_code)]
 pub struct SignerDetails {
     pub account: AccountId32,
     pub name: String,

--- a/src/runtimes/kusama.rs
+++ b/src/runtimes/kusama.rs
@@ -1493,7 +1493,7 @@ pub async fn try_fetch_pool_operators_for_compound(
 
                     let claimable: u128 = Decode::decode(&mut &*bytes)?;
 
-                    if claimable > config.pool_compound_threshold.into() {
+                    if claimable > config.pool_compound_threshold as u128 {
                         members.push(pool.roles.depositor.clone());
                     }
                 }
@@ -1565,7 +1565,7 @@ pub async fn try_fetch_pool_members_for_compound(
 
                     let claimable: u128 = Decode::decode(&mut &*bytes)?;
 
-                    if claimable > config.pool_compound_threshold.into() {
+                    if claimable > config.pool_compound_threshold as u128 {
                         members.push(member);
                     }
                 }

--- a/src/runtimes/paseo.rs
+++ b/src/runtimes/paseo.rs
@@ -1487,7 +1487,7 @@ pub async fn try_fetch_pool_operators_for_compound(
 
                     let claimable: u128 = Decode::decode(&mut &*bytes)?;
 
-                    if claimable > config.pool_compound_threshold.into() {
+                    if claimable > config.pool_compound_threshold as u128 {
                         members.push(pool.roles.depositor.clone());
                     }
                 }
@@ -1559,7 +1559,7 @@ pub async fn try_fetch_pool_members_for_compound(
 
                     let claimable: u128 = Decode::decode(&mut &*bytes)?;
 
-                    if claimable > config.pool_compound_threshold.into() {
+                    if claimable > config.pool_compound_threshold as u128 {
                         members.push(member);
                     }
                 }

--- a/src/runtimes/polkadot.rs
+++ b/src/runtimes/polkadot.rs
@@ -1494,7 +1494,7 @@ pub async fn try_fetch_pool_operators_for_compound(
 
                     let claimable: u128 = Decode::decode(&mut &*bytes)?;
 
-                    if claimable > config.pool_compound_threshold.into() {
+                    if claimable > config.pool_compound_threshold as u128 {
                         members.push(pool.roles.depositor.clone());
                     }
                 }
@@ -1566,7 +1566,7 @@ pub async fn try_fetch_pool_members_for_compound(
 
                     let claimable: u128 = Decode::decode(&mut &*bytes)?;
 
-                    if claimable > config.pool_compound_threshold.into() {
+                    if claimable > config.pool_compound_threshold as u128 {
                         members.push(member);
                     }
                 }

--- a/src/runtimes/westend.rs
+++ b/src/runtimes/westend.rs
@@ -1481,7 +1481,7 @@ pub async fn try_fetch_pool_operators_for_compound(
 
                     let claimable: u128 = Decode::decode(&mut &*bytes)?;
 
-                    if claimable > config.pool_compound_threshold.into() {
+                    if claimable > config.pool_compound_threshold as u128 {
                         members.push(pool.roles.depositor.clone());
                     }
                 }
@@ -1553,7 +1553,7 @@ pub async fn try_fetch_pool_members_for_compound(
 
                     let claimable: u128 = Decode::decode(&mut &*bytes)?;
 
-                    if claimable > config.pool_compound_threshold.into() {
+                    if claimable > config.pool_compound_threshold as u128 {
                         members.push(member);
                     }
                 }

